enterprise/internal/authz/github/app_test.go

```diff
@@ -0,0 +1,72 @@
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/google/go-github/v31/github"
+
+	"github.com/sourcegraph/sourcegraph/internal/types"
+	"github.com/sourcegraph/sourcegraph/schema"
+)
+
+func TestNewAppProvider(t *testing.T) {
+	t.Run("test new app provider client", func(t *testing.T) {
+		srvHit := false
+		srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if r.URL.Path == "/api/v3/app/installations/1234/access_tokens" {
+				tokenString := "1234"
+				tokenExpiry := time.Now()
+				token := github.InstallationToken{
+					Token:        &tokenString,
+					ExpiresAt:    &tokenExpiry,
+					Permissions:  &github.InstallationPermissions{},
+					Repositories: []*github.Repository{},
+				}
+
+				respJson, _ := json.Marshal(token)
+
+				w.Header().Set("Content-Type", "application/json")
+				w.Write(respJson)
+			} else {
+				srvHit = true
+				gotHeader := r.Header.Get("Authorization")
+				wantHeader := "Bearer 1234"
+				if gotHeader != wantHeader {
+					t.Fatalf("wanted authorization header %s, got %s", wantHeader, gotHeader)
+				}
```

Comment:
We're trading towards using `assert` and `require` packages.

```suggestion
				assert.Equal(t, wantHeader, gotHeader)
```